engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectAndViewAnalyzer.java

```diff
@@ -189,6 +222,12 @@ public static SelectAndViewAnalyzerWrapper create(
                     final WritableColumnSource<?> scs =
                             flatResult || flattenedResult ? sc.newFlatDestInstance(targetDestinationCapacity)
                                     : sc.newDestInstance(targetDestinationCapacity);
+
+                    if (flattenedResult && realColumn != null && !resultColumns.contains(realColumn.getSourceName())) {
+                        // this source column a candidate for preservation if referenced again
+                        resultAlias.put(realColumn.getSourceName(), scs);
```

Comment:
Why is the source column a candidate for preservation and not the destination name?

Reply:
I think the optimization holds true for any preserved column, no matter the mode.  It is a bit odd to make copies of a column, as you are not adding any meaningful data.  If you were trying to capture some kind of stateful function it would change the answer; but I think if you do that maybe you get what you deserve.